src/java/org/apache/cassandra/cql3/terms/Constants.java

```diff
@@ -489,8 +489,10 @@ public void execute(DecoratedKey partitionKey, UpdateParameters params) throws I
             else if (column.type instanceof NumberType<?>)
             {
                 @SuppressWarnings("unchecked") NumberType<Number> type = (NumberType<Number>) column.type;
-                ByteBuffer increment = t.bindAndGet(params.options);
-                ByteBuffer current = getCurrentCellBuffer(partitionKey, params);
+                ByteBuffer increment = type.sanitize(t.bindAndGet(params.options));
```

Comment:
`type.sanitize` is a bug fix for empty bytes.  Numeric types allow `byte[0]` and its semantic is weird... in 90% of the codebase its `null` but in that 10% its not... this code path hits a NPE as its part of the 90%...

empty is only not null if and only if you do `col = <empty>`, any other path its `null`...